imitation/trainer.py

```diff
@@ -1,69 +1,90 @@
-from typing import Optional
+from typing import Dict, Optional, Sequence, Union
 from warnings import warn
 
+import gym
 import numpy as np
+from stable_baselines.common.base_class import BaseRLModel
 from stable_baselines.common.vec_env import VecEnvWrapper
 import tensorflow as tf
 from tqdm import tqdm
 
-import imitation.summaries as summaries
-import imitation.util as util
+from imitation import summaries, util
+from imitation.discrim_net import DiscrimNet
 from imitation.util.buffer import ReplayBuffer
 
 
 class Trainer:
-
-  def __init__(self, env, gen_policy, discrim, expert_policies, *,
-               n_disc_samples_per_buffer=200, n_expert_samples=4000,
+  """Trainer for GAIL and AIRL."""
+
+  def __init__(self,
+               env: Union[gym.Env, str],
+               gen_policy: BaseRLModel,
+               discrim: DiscrimNet,
+               expert_policies: Sequence[BaseRLModel], *,
+               disc_opt_cls: tf.train.Optimizer = tf.train.AdamOptimizer,
+               disc_opt_kwargs: Dict = {},
```

Comment:
It's considered better form, I think, to use `dict` when using the generic properties of `Dict`

Reply:
:+1: did not know this, makes sense